chainer/functions/array/expand_dims.py

```diff
@@ -0,0 +1,39 @@
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class ExpandDims(function.Function):
+
+    """Expands dimenstions of an input array without copy."""
+
+    def __init__(self, axis):
+        self.axis = int(axis)
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+        if self.axis >= 0:
+            type_check.expect(x_type.ndim >= self.axis)
+        else:
+            type_check.expect(x_type.ndim >= -self.axis - 1)
+
```

Comment:
Both NumPy and CuPy's `expand_dims` can accept `axis` that does not satisfy `-ndim-1 <= axis <= ndim`. If either of inequalities does not hold, the value of `axis` is regarded as an edge value.